Froger/MainLoop.java

```diff
@@ -0,0 +1,121 @@
+package Froger;
+
+import javafx.animation.Animation;
+import javafx.animation.KeyFrame;
+import javafx.animation.Timeline;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.ImagePattern;
+import javafx.scene.text.Font;
+import javafx.stage.Stage;
+import javafx.util.Duration;
+
+
+public class MainLoop extends Application {
+
+    private int score = 0;
+    private int sizeMapX = 20;
+    private int sizeMapY = 10;
+    private Group root = new Group();
+    private GridMap gridMap = new GridMap(sizeMapX, sizeMapY);
+    private Frog frog;
+    private Cars cars[];
+    private Gold gold;
+    private Timeline tl;
+    private HealthPoints healthPoints;
+    private Label lbScore;
+
+    @Override
+    public void start(Stage primaryStage) {
+        Scene scene = new Scene(root, 700, 700, Color.WHITE);
+        healthPoints = new HealthPoints();
+        frog = new Frog(gridMap);
+        cars = new Cars[sizeMapY-1];
+        gold = new Gold(gridMap);
+
+        addGridMap();
+        addCars();
+        addScoreLabel();
+        changeBackground(scene);
+
+        //timelines
+        collision();
+        catchMoney();
+
+        scene.setOnKeyPressed(keyEvent -> {
+            frog.jump(keyEvent.getCode());
+            frog.setVisible(true);
+        });
+
+        root.getChildren().addAll(gridMap, healthPoints, healthPoints.getInscriptionHp(), lbScore);
+
+        primaryStage.setTitle("Frogger");
+        primaryStage.setScene(scene);
+        primaryStage.setResizable(false);
+        primaryStage.show();
+    }
+
+    private void addCars(){
+        for (int i = 1; i < sizeMapY-1; i++) {
+            cars[i] = new Cars(20, 20, Color.RED, gridMap);
```

Comment:
ktos madry kiedys powiedzial ze jak masz zrobic kopiuj wklej to lepiej zrobic funkcje
wiec zamiast  tego:
```java
            cars[i] = new Cars(20, 20, Color.RED, gridMap);
            cars[i].setLocation(1, i);
            cars[i].run();
            cars[i] = new Cars(20, 20, Color.VIOLET, gridMap);
            cars[i].setLocation(1, i);
            cars[i].run();
```
o wiele ladniej wygladalo by :
```java
cars[i] = initNewCar(Color.RED, i, gridMap)
cars[i] = initNewCar(Color.VIOLET, i, gridMap)
```
czy cos w tym stylu
Jak piszesz w intellij to ``CTRL`` + ``ALT`` + ``M`` bedzie Twoim przyjacielem